hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -201,16 +201,16 @@ public Block getStoredBlock(String bpid, long blkid)
    * The deepCopyReplica call doesn't use the datasetock since it will lead the
    * potential deadlock with the {@link FsVolumeList#addBlockPool} call.
    */
+  @SuppressWarnings("unchecked")
   @Override
   public Set<? extends Replica> deepCopyReplica(String bpid)
       throws IOException {
-    Set<? extends Replica> replicas = null;
+    Set<? extends Replica> replicas;
```

Comment:
can this change relate to trunk also..?  Even this can be emptyset

Reply:
Thanks @brahmareddybattula 
I think we can apply some changes to trunk given that the type casting is unchecked and causes some Javac warnings.
Perhaps we can file another refactoring jira to address these issues on trunk.

Reply:
can please raise jira to track this..?